File: app/src/main/java/su/linka/linkapaperboard/SlideButtonsController.java
package su.linka.linkapaperboard;

import android.support.v7.widget.AppCompatButton;
import android.text.TextUtils;
import android.view.View;

public class SlideButtonsController {

    private final AppCompatButton leftBtn;
    private final AppCompatButton rightBtn;


    public SlideButtonsController(View view) {
        leftBtn = view.findViewById(R.id.left_btn);
        rightBtn = view.findViewById(R.id.right_btn);
    }

    public void setTextForLeftBtn(String text) {
        setTextForButton(leftBtn, text, "⬅️");
    }

    public void setTextForRightBtn(String text) {
        setTextForButton(rightBtn, text, "➡️");
    }

    private void setTextForButton(AppCompatButton button, String text, String arrowSymbol) {
        if (TextUtils.isEmpty(text)) {
            return;
        }
        int length = text.length();
        StringBuilder stringBuilder = new StringBuilder();
        if (length == 1) {
            stringBuilder.append(text);
        } else {
            stringBuilder.append(text.substring(0, 3))
                    .append("...")
                    .append(text.substring(length - 3));
        }
        stringBuilder.append("\n")
                .append(arrowSymbol);

        button.setText(stringBuilder.toString());
    }

}

File: app/src/main/java/su/linka/linkapaperboard/SlideButtonsController.kt
package su.linka.linkapaperboard

import androidx.appcompat.widget.AppCompatButton
import android.view.View

class SlideButtonsController(view: View) {

    private val leftBtn: AppCompatButton = view.findViewById(R.id.left_btn)
    private val rightBtn: AppCompatButton = view.findViewById(R.id.right_btn)

    fun setTextForLeftBtn(text: String) {
        setTextForButton(leftBtn, text, "⬅️")
    }

    fun setTextForRightBtn(text: String) {
        setTextForButton(rightBtn, text, "➡️")
    }

    private fun setTextForButton(button: AppCompatButton, text: String?, arrowSymbol: String) {
        if (text.isNullOrEmpty()) {
            return
        }
        button.text = buildButtonText(text, arrowSymbol)
    }

    companion object {
        internal fun buildButtonText(text: String, arrowSymbol: String): String {
            val formatted = if (text.length == 1) {
                text
            } else {
                "${text.take(3)}...${text.takeLast(3)}"
            }
            return "$formatted\n$arrowSymbol"
        }
    }
}
